Dashboard redesign

pratishshr opened the pull request:
- Redesign dashboard
 - Clean design
 - Interchangeable header logo

* Need a default chill logo

pratishshr commented:
We should probably use .yml instead of .env here as well.

mesaugat commented on src/routes.js:
```diff
@@ -1,27 +1,23 @@
+import Dashboard from './components/Dashboard';
+import Header from './components/commons/Header';
+import Footer from './components/commons/Footer';
 
```
Maybe destructure here: `const { app } = config;`

mesaugat commented on src/routes.js:
```diff
@@ -1,27 +1,23 @@
-        </Switch>
-      </div>
+    <div>
+      <Header logoUrl={config.app.logoUrl} style={{ height: config.app.logoHeight }} />
```
And `app.logoUrl`, `app.logoHeight`

mesaugat commented on src/config.js:
```diff
@@ -1,4 +1,8 @@
 module.exports = {
```
export default?

kabirbaidhya commented:
@pratishshr I think yes, then the user doesn't have to bother about multiple config files. Just one `chill.yml` would do. For now it's okay, can do that later.

mesaugat commented on src/components/commons/Spinner.js:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const Spinner = () => {
```
How about:

```js
const Spinner = () => (
  <div className="loader">Loading...</div>
);
```

mesaugat commented on src/components/commons/Panel.js:
```diff
@@ -0,0 +1,20 @@
+
```
Remove new line

mesaugat commented on src/components/Dashboard/StatusList.js:
```diff
@@ -0,0 +1,133 @@
+   * @returns {Boolean}
+   */
+  isOperational(services) {
+    for (let key in services) {
```
```js
for (let service of services)
```

mesaugat commented on src/components/Dashboard/StatusList.js:
```diff
@@ -0,0 +1,133 @@
+  }
+
+  /**
+   * Fetch list of services 
```
Full stop.

kabirbaidhya commented on .env.example:
```diff
@@ -1,20 +1,7 @@
-APP_PORT='8000'
-APP_HOST='127.0.0.1'
+# APP
+APP_LOGO={APP_LOGO}
```
Nice :+1:

kabirbaidhya commented on src/components/commons/Footer.js:
```diff
@@ -0,0 +1,21 @@
+            className="footer-text"
+            href="https://github.com/leapfrogtechnology/chill"
+          >
+            Powered By Chill
```
Could it be `Powered by <a href...>Chill</a>`.?

kabirbaidhya commented on src/components/Dashboard/StatusList.js:
```diff
@@ -0,0 +1,133 @@
+import Panel from '../commons/Panel';
+import Spinner from '../commons/Spinner';
+
+const STATUS_UP_CLASS = 'status-up';
```
Can these constants be exported from the service `'../../services/status'`? All UI agnostic code should go to services.

kabirbaidhya commented on src/components/Dashboard/StatusList.js:
```diff
@@ -0,0 +1,133 @@
+        title={message}
+        titleClass={titleClass}
+      >
+        <ul className="list-group">
```
Can we extract this whole `<ul>`block as a separate functional component `ServiceList`? Better separation of concern.

pratishshr commented:
- Added tooltip
- Added logic for showing 3 different system outage messages
  - If any 1 down - Partial outage
  - If all down - Major outage 
  - If none - All systems operational

<img width="944" alt="screen shot 2017-06-28 at 11 28 31 am" src="https://user-images.githubusercontent.com/13301510/27622216-6c513c30-5bf5-11e7-878e-18cd83f4bf58.png">
<img width="897" alt="screen shot 2017-06-28 at 11 29 00 am" src="https://user-images.githubusercontent.com/13301510/27622219-70b19748-5bf5-11e7-8267-00a49fdbcd77.png">
<img width="892" alt="screen shot 2017-06-28 at 11 29 33 am" src="https://user-images.githubusercontent.com/13301510/27622222-749fd25c-5bf5-11e7-9976-f5e169b80226.png">

kabirbaidhya commented on src/constants/statusConstants.js:
```diff
@@ -0,0 +1,14 @@
+const statusConstants = {
+  STATUS_UP: 'up',
```
instead of `default` export here. Export each constant. 
Like this:
```js
export const STATUS_UP = 'up';
export const STATUS_DOWN: 'down',
...
```

kabirbaidhya commented on src/enums/Outage.js:
```diff
@@ -0,0 +1,7 @@
+const Outage = {
+  NONE: 0,
```
Same here `export` each constant

kabirbaidhya commented on src/services/status.js:
```diff
@@ -39,3 +38,68 @@ export function getServiceCountsByStatus(services) {
+ * Check outage status.
+ * 
+ * @param {Array} services 
+ * @returns {Outage}
```
`Outage` is not a class or a type function. It's just a numeric data type as we don't have Enum data structures on JavaScript

kabirbaidhya commented on src/services/status.js:
```diff
@@ -1,14 +1,13 @@
 import config from '../config';
 import http from '../utils/http';
+import Outage from '../enums/Outage';
```
camelCase naming convention unless it's a class or Component.

kabirbaidhya commented:
One suggestion about text change. Instead of "Partial Outage" can we show some message like "3/10 Systems Operational"?
@leapfrogtechnology/the-chill-team

kabirbaidhya commented on src/components/commons/Panel.js:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Panel = ({ title, children, className = 'status-up' }) => (
```
We now have a constant for `'status-up'` class right?

romitamgai commented on src/config.js:
```diff
@@ -1,4 +1,8 @@
-module.exports = {
+export default {
+  app: {
+    logoUrl: process.env.APP_LOGO,
```
could we add fallback for APP_LOGO?

romitamgai replied:
could we add fallback for APP_LOGO?

kabirbaidhya replied:
Yes, the chill logo :+1:

deepsadhi commented on public/css/spinner.css:
```diff
@@ -0,0 +1,57 @@
+  -webkit-animation: load7 1.8s infinite ease-in-out;
+  animation: load7 1.8s infinite ease-in-out;
+}
+.loader {
```
Add one line above

deepsadhi commented on public/css/spinner.css:
```diff
@@ -0,0 +1,57 @@
+  -webkit-animation-delay: -0.16s;
+  animation-delay: -0.16s;
+}
+.loader:before,
```
Add one line above

deepsadhi commented on public/css/spinner.css:
```diff
@@ -0,0 +1,57 @@
+.loader:after {
+  left: 3.5em;
+}
+@-webkit-keyframes load7 {
```
Add one line above

deepsadhi commented on public/css/spinner.css:
```diff
@@ -0,0 +1,57 @@
+    box-shadow: 0 2.5em 0 0;
+  }
+}
+@keyframes load7 {
```
Add one line above
